webrender/res/ps_border_corner.glsl

```diff
@@ -110,10 +110,10 @@ void write_color(vec4 color0, vec4 color1, int style, vec2 delta, int instance_k
             break;
     }
 
-    vColor00 = vec4(color0.rgb * modulate.x, color0.a);
-    vColor01 = vec4(color0.rgb * modulate.y, color0.a);
-    vColor10 = vec4(color1.rgb * modulate.z, color1.a);
-    vColor11 = vec4(color1.rgb * modulate.w, color1.a);
+    vColor00 = vec4(clamp(color0.rgb * modulate.x, vec3(0.0), vec3(1.0)), color0.a);
```

Comment:
wouldn't it be simpler to clamp the `modulate` vec?

Reply:
It's not necessarily incorrect for modulate to be out of the 0..1 range if we want the modulation to brighten the color. I am not sure whether we need that (I assume we do, otherwise maybe the bug is in the values we pass to the shader)

Reply:
right, I see now. Why are you clamping the colors here anyway?

Reply:
Under the assumption that we can pass modulate components greater than 1, this is the place that makes most sense to me since this is where the base colors are computed. Any later computation can "get it wrong" by assuming they are in the range that goes according to their semantic (colors). Where would you have done the clamping?

Reply:
It largely depends on whether the semantics of border colors is defined for vertices as opposed to pixels. I assumed that W3C would care about pixel colors, while the fact that GPU have vertices and interpolate between them is an implementation detail. In this case, the clamping should be done for the pixels, not here.

Reply:
These are flat varying variables so there's no interpolation at all. Even if there was, it would be incorrect to interpolate between the the non-clamped values (you'd end up with colors constant over a certain range as they are capped at the brightest value until the result of the interpolation gets to the correct range and then it would start interpolating. This is actually the equivalent of one of the bugs this PR is fixing but at the vertex interpolation level instead of the mix between two sides of the border.